### ISSUE TITLE
Style/RedundantReturn doesn't detect offenses if an ensure block is present

### ISSUE THREAD

**Issue by jturkel** [bbatsov/rubocop#5401]
The Style/RedundantReturn cop doesn't detect offenses if an `ensure` block is present in a method. 

--------

## Expected behavior

Redundant returns should be flagged as an offense even if the method has an `ensure` block.

## Actual behavior

No offenses are detected.

## Steps to reproduce the problem

The following code should trigger an offense:

```ruby
class Foo
  def bar
    return nil
  ensure
    nil
  end
end
```

## RuboCop version

```
$ rubocop -V
0.52.1
```

**Comment by asherkach** [comment]
This seems to be a generalization of [this comment](https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/cop/style/redundant_return.rb#L21-L22) in the cop that indicates the cop should be generalized to cover a wider footprint.

**Comment by asherkach** [comment]
Fix (this and related issues) forthcoming...
